### DISCUSSION TITLE
Panel Loading

### DISCUSSION THREAD

**Discussion by CodingPupper3033** [inventree/InvenTree#5869] [Q&A]
Hello!

If you're not sick of me yet, I have another question to ask! (Thanks for your help guys, I am very grateful)

I am trying to recreate a screen similar to how the base Inventree does it. It appears you guys make bootstrap panels inside {% page_content %}. That makes sense to me. However, it also seems that you initially hide the panels; I'd assume you're waiting for the data to load. However, I need help figuring out how to unhide the panels. How do you "load the panels"? I found the `onPanelLoad` function but not how to call it. 

I don't see any special functions that call the panel load in the templates; the only thing I can see is the `{{ block.super }}` which I have tried calling inside the `{% block js_ready %}`.

I can't find the file anymore, but from what I remember, I thought I saw a function that searches for the class "panel-hidden" and loads it, but it doesn't seem to be calling on my panels.

Additionally, here's something I have been struggling with but need to look into more: How do you generate a table? It appears in the javascript you call a function that ultimately calls table.generateBootstrapTable. Is that the recommended way to create tables similar to those in inventree?

**Comment by CodingPupper3033** [comment]
Panel loading is explained in pull request #5896 and can be accessed [here](https://docs.inventree.org/en/latest/extend/plugins/urls/)

I'm still trying to figure out how to make a table, I'll hopefully answer that soon!

**Comment by SergeoLacruz** [comment]
I created a table here:

[example](https://github.com/SergeoLacruz/inventree-supplier-panel/blob/master/inventree_supplier_panel/templates/supplier_panel/mouser.html)

There must be better solutions but it is OK for me so far.
